rfc#00003

```diff
@@ -0,0 +1,213 @@
+# Title (Ex. 00002: Trust domains and Enarx)
+- Authors: [Mike Bursell](you@github-email)
+- Status: [PROPOSED](/README.md#proposed)
+- Since: 2020-02-13
+- Status Note: under discussion
+- Start Date: 2020-02-13 (date you started working on this idea)
+- Tags: trust
+- Addresses: (Issue #192)[https://github.com/enarx/enarx/issues/192]
+
+## Summary
+
+Trust domains are a key concept within Enarx.  This RFC provides an
+introduction to trust domains in Enarx.
+
+## Motivation
+
+In order to explain the motivations behind Enarx, our design principles
+and aims, we need to be able to explain what trust domains are.  This RFC
+attempts to provide an introduction.
+
+## Tutorial
+
+See rfc#00002 for an introduction to trust relationships.
+
+### Applying trust relationships to Enarx
+The key trust relationship for Enarx is that held by the tenant
+(workload owner) to that workload.  Enarx aims to provide assurances to
+the tenant that the integrity and confidentiality of that workload are
+protected: these are the key contexts of that trust, and actions can be
+assured if they are maintained.  Trust may decay or be built up in the
+workload - the time context.  In this case, the workload does not have an
+equivalent set of assurances about the tenant, so the relationship is
+not symmetrical.
+
+See rfc#00002 for an introduction to trust relationships.
+
+### Applying trust domains to Enarx - default model
+There are various possible models for trust domains in an Enarx
+deployment - this section describes the simplest, and default model.
+In the default Enarx model, we start off by defining the following parties
+- tenant
+- host owner (the host operator may be separate, but is considered to
+be the same for this description)
+- entities running processes on the host (includes other workloads
+and compromised processes)
+- man-in-the-middle entities
+Clearly, the first two are the key actors - the others are included for
+reference (no threat model is not defined here).
+
+The following components are defined:
+- host 
+- host CPU + CPU firmware
+- Orchestrator (e.g. Kubernetes, Openshift, Openstack)
+- Tenant workload image
+- Tenant workload image repository
+- Enarx client agent (ECA)
+- Enarx host agent (EHA)
+- Enarx runtime image
+- TEE instance
+- Empty Keep
+- Provisioned Keep
+- Tenant workload
+- Other workloads
+Not all of these exist at all points in the process.
+
+
+### Non-Enarx-enabled use case states
+In the non-TEE-assisted case, all processes running on the host are in
+the host's trust domain (this assumes no compromised processes).  We
+ignore man-in-the-middle entities.  For the default trust domain model,
+we assume that the Orchestrator is operated by the tenant and that the
+Tenant workload image repository is also operated by the tenant.
+
+#### State 0 non-Enarx-enabled use case (initial)
+No Enarx-related components are relevant to this use case.
```

Comment:
```suggestion
Note: no Enarx-related components are relevant to this use case.

The following components are considered:
```

Reply:
Not a note: it's a core descriptive point.